### PR TITLE
feature: Add support for path in file elements on Clover format [CY-2872]

### PR DESCRIPTION
**Pull request by rtfpessoa**

### CODE REVIEW

File: src/main/scala/com/codacy/parsers/implementation/CloverParser.scala [hunk 1]
```diff
@@ -43,9 +44,23 @@ object CloverParser extends CoverageParser with XmlReportParser {
...
-    val filename = fileNode \@ "name"
-    val cleanFileName = TextUtils.sanitiseFilename(filename).stripPrefix(rootPath).stripPrefix("/")
+  private def getCoverageFileReport(rootPath: String, fileNode: Node): CoverageFileReport = {
+    val filePath = fileNode.attribute("path").flatMap(_.headOption.map(_.text)).map(TextUtils.sanitiseFilename)
+    val filename = fileNode.attribute("name").flatMap(_.headOption.map(_.text)).map(TextUtils.sanitiseFilename)
+
+    val relativeFilePath = filePath
+      .orElse(filename)
+      .fold[String] {
+        throw new Exception(
```
Comment (rtfpessoa) [inline src/main/scala/com/codacy/parsers/implementation/CloverParser.scala:54]: I am still looking at how to do this error better.

File: src/main/scala/com/codacy/parsers/implementation/CloverParser.scala [hunk 2]
```diff
@@ -43,9 +44,23 @@ object CloverParser extends CoverageParser with XmlReportParser {
...
+          "Could not read file path due to missing 'path' and 'name' attributes in the report file element."
```
Comment (afsmeira) [inline src/main/scala/com/codacy/parsers/implementation/CloverParser.scala:55]: Minor nitpick: the message should actually be "(...) due to missing 'path' **or** 'name' attributes (...)"

**Review by afsmeira — DISMISSED** [review]

File: src/main/scala/com/codacy/parsers/implementation/CloverParser.scala [hunk 3]
```diff
@@ -43,9 +44,23 @@ object CloverParser extends CoverageParser with XmlReportParser {
...
   }
 
-  private def getCoverageFileReport(rootPath: String, fileNode: Node) = {
-    val filename = fileNode \@ "name"
-    val cleanFileName = TextUtils.sanitiseFilename(filename).stripPrefix(rootPath).stripPrefix("/")
+  private def getCoverageFileReport(rootPath: String, fileNode: Node): CoverageFileReport = {
+    val filePath = fileNode.attribute("path").flatMap(_.headOption.map(_.text)).map(TextUtils.sanitiseFilename)
+    val filename = fileNode.attribute("name").flatMap(_.headOption.map(_.text)).map(TextUtils.sanitiseFilename)
+
+    val relativeFilePath = filePath
```
Comment (rtfpessoa) [inline src/main/scala/com/codacy/parsers/implementation/CloverParser.scala:51]: I could not find a spec for the format so I honestly cannot understand if this is the best solution :cry:

File: src/main/scala/com/codacy/parsers/implementation/CloverParser.scala (hunk 2 above)
Comment (rtfpessoa) [reply src/main/scala/com/codacy/parsers/implementation/CloverParser.scala:55]: That is a good question. I had or initially, but changed.
We just need one of them, so I want to say they both were missing. Doesn't and make more sense?

Comment (afsmeira) [reply src/main/scala/com/codacy/parsers/implementation/CloverParser.scala]: If you only need on, then `or` definitely makes more sense to me. Anyway, this is just a nitpick, so I'll approve anyway.

**Review by afsmeira — APPROVED** [review]

**Pull request merged**